Remove non-standard `KeyboardEvent.keyLocation`

Ahmad-S792 opened the pull request:
#### bba680e17fee76325be7facc38899eda3f43d419
<pre>
Remove non-standard `KeyboardEvent.keyLocation`

<a href="https://bugs.webkit.org/show_bug.cgi?id=267819">https://bugs.webkit.org/show_bug.cgi?id=267819</a>

Reviewed by Ryosuke Niwa.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium
by removing non-standard KeyboardEvent.keyLocation (alias of &apos;location&apos;).

It is not in web-specification [1].

[1] <a href="https://w3c.github.io/uievents/#events-keyboardevents">https://w3c.github.io/uievents/#events-keyboardevents</a>

It was never shipped in Gecko / Firefox while removed in Blink
in 2016 and with commit [2]:

[2] <a href="https://chromium.googlesource.com/chromium/src.git/+/263d94f9b2dc4a0a3cf5b3124c1cc8d2163a4410">https://chromium.googlesource.com/chromium/src.git/+/263d94f9b2dc4a0a3cf5b3124c1cc8d2163a4410</a>

* Source/WebCore/dom/KeyboardEvent.cpp:
(KeyboardEvent::KeyboardEvent):
* Source/WebCore/dom/KeyboardEvent.h:
* Source/WebCore/dom/KeyboardEvent.idl:
* LayoutTests/fast/events/constructors/keyboard-event-constructor.html: Rebaselined
* LayoutTests/fast/events/constructors/keyboard-event-constructor-expected.txt: Rebaselined
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-control-expected.txt: Rebaselined
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-control-option-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-control-shift-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-meta-control-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-meta-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-meta-option-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-meta-shift-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-option-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-option-shift-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-shift-expected.txt: Ditto

Canonical link: <a href="https://commits.webkit.org/273457@main">https://commits.webkit.org/273457@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/ccfe6cc83a9f45b44594648632966b5d61404e5b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35436 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14363 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/37556 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/38171 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/31951 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/36630 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/16787 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11406 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/30785 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/35987 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/12178 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31559 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/10648 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/10712 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/31678 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39419 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/32240 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/32009 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/36649 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/10849 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/8759 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/34697 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/12592 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/31339 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8108 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11373 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/11652 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on LayoutTests/fast/events/constructors/keyboard-event-constructor.html:
```diff
@@ -112,8 +112,7 @@
 });
-shouldBe("new KeyboardEvent('eventType', { keyLocation: KeyboardEvent.DOM_KEY_LOCATION_LEFT }).location", "KeyboardEvent.DOM_KEY_LOCATION_LEFT");
-shouldBe("new KeyboardEvent('eventType', { location: KeyboardEvent.DOM_KEY_LOCATION_LEFT }).keyLocation", "KeyboardEvent.DOM_KEY_LOCATION_LEFT");
+shouldBe("new KeyboardEvent('eventType', { location: KeyboardEvent.DOM_KEY_LOCATION_LEFT }).location", "KeyboardEvent.DOM_KEY_LOCATION_LEFT");
```
Not really asserting anything useful here anymore. Previously this made sure `keyLocation` and `location` were really referring to the same data. This can be removed.

Ahmad-S792 replied:
Thanks. Done in [3c3bc33](https://github.com/WebKit/WebKit/commit/3c3bc33da04b37943b123aaf42fa138c8389ffe5).

aprotyas commented on LayoutTests/fast/events/constructors/keyboard-event-constructor.html:
```diff
@@ -80,8 +80,8 @@
-// location / keyLocation is passed.
-["location", "keyLocation"].forEach(function (propertyName) {
+// location is passed.
+["location"].forEach(function (propertyName) {
```
Entirely optional, but there's no need for a `forEach` construct here anymore since it's just a single property.

<details>
<summary>Reduce complexity in the test case</summary>

```diff
diff --git a/LayoutTests/fast/events/constructors/keyboard-event-constructor.html b/LayoutTests/fast/events/constructors/keyboard-event-constructor.html
index 0b381c4db3e0..09761747e7a2 100644
--- a/LayoutTests/fast/events/constructors/keyboard-event-constructor.html
+++ b/LayoutTests/fast/events/constructors/keyboard-event-constructor.html
@@ -80,40 +80,36 @@ shouldBeEqualToString("new KeyboardEvent('eventType', { keyIdentifier: [1, 2, 3]
 shouldBeEqualToString("new KeyboardEvent('eventType', { keyIdentifier: {koakuma: 12345} }).keyIdentifier", "[object Object]");
 shouldBeEqualToString("new KeyboardEvent('eventType', { keyIdentifier: {valueOf: function () { return 'koakuma'; } } }).keyIdentifier", "[object Object]");

-// location / keyLocation is passed.
-["location", "keyLocation"].forEach(function (propertyName) {
-    // numbers within the unsigned long range.
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": KeyboardEvent.DOM_KEY_LOCATION_STANDARD })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": KeyboardEvent.DOM_KEY_LOCATION_LEFT })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_LEFT");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": 4294967294 })." + propertyName, "4294967294");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": 4294967295 })." + propertyName, "4294967295");
-
-    // numbers out of the unsigned long range.
-    // 2^{53}-1, the largest number that can be exactly represented by double.
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": 9007199254740991 })." + propertyName, "4294967295");
-    // 2^{64}-1
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": 18446744073709551615 })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": 12345678901234567890 })." + propertyName, "3944679424");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": -1 })." + propertyName, "4294967295");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": 123.45 })." + propertyName, "123");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": NaN })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
-
-    // Non-numeric values.
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": undefined })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": null })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": '' })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": '12345' })." + propertyName, "12345");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": '' })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": '12345' })." + propertyName, "12345");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": '12345a' })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": 'abc' })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": [] })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": [12345] })." + propertyName, "12345");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": [12345, 67890] })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": {} })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": {moemoe: 12345} })." + propertyName, "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
-    shouldBe("new KeyboardEvent('eventType', { " + propertyName + ": {valueOf: function () { return 12345; }} })." + propertyName, "12345");
-});
-shouldBe("new KeyboardEvent('eventType', { keyLocation: KeyboardEvent.DOM_KEY_LOCATION_LEFT }).location", "KeyboardEvent.DOM_KEY_LOCATION_LEFT");
-shouldBe("new KeyboardEvent('eventType', { location: KeyboardEvent.DOM_KEY_LOCATION_LEFT }).keyLocation", "KeyboardEvent.DOM_KEY_LOCATION_LEFT");
+// location is passed.
+// numbers within the unsigned long range.
+shouldBe("new KeyboardEvent('eventType', { location : KeyboardEvent.DOM_KEY_LOCATION_STANDARD }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
+shouldBe("new KeyboardEvent('eventType', { location : KeyboardEvent.DOM_KEY_LOCATION_LEFT }).location", "KeyboardEvent.DOM_KEY_LOCATION_LEFT");
+shouldBe("new KeyboardEvent('eventType', { location : 4294967294 }).location", "4294967294");
+shouldBe("new KeyboardEvent('eventType', { location : 4294967295 }).location", "4294967295");
+
+// numbers out of the unsigned long range.
+// 2^{53}-1, the largest number that can be exactly represented by double.
+shouldBe("new KeyboardEvent('eventType', { location : 9007199254740991 }).location", "4294967295");
+// 2^{64}-1
+shouldBe("new KeyboardEvent('eventType', { location : 18446744073709551615 }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
+shouldBe("new KeyboardEvent('eventType', { location : 12345678901234567890 }).location", "3944679424");
+shouldBe("new KeyboardEvent('eventType', { location : -1 }).location", "4294967295");
+shouldBe("new KeyboardEvent('eventType', { location : 123.45 }).location", "123");
+shouldBe("new KeyboardEvent('eventType', { location : NaN }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
+
+// Non-numeric values.
+shouldBe("new KeyboardEvent('eventType', { location : undefined }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
+shouldBe("new KeyboardEvent('eventType', { location : null }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
+shouldBe("new KeyboardEvent('eventType', { location : '' }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
+shouldBe("new KeyboardEvent('eventType', { location : '12345' }).location", "12345");
+shouldBe("new KeyboardEvent('eventType', { location : '12345a' }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
+shouldBe("new KeyboardEvent('eventType', { location : 'abc' }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
+shouldBe("new KeyboardEvent('eventType', { location : [] }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
+shouldBe("new KeyboardEvent('eventType', { location : [12345] }).location", "12345");
+shouldBe("new KeyboardEvent('eventType', { location : [12345, 67890] }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
+shouldBe("new KeyboardEvent('eventType', { location : {} }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
+shouldBe("new KeyboardEvent('eventType', { location : {moemoe: 12345} }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
+shouldBe("new KeyboardEvent('eventType', { location : {valueOf: function () { return 12345; }} }).location", "12345");
```

</details>

Ahmad-S792 replied:
Thanks. Done in [3c3bc33](https://github.com/WebKit/WebKit/commit/3c3bc33da04b37943b123aaf42fa138c8389ffe5).

aprotyas reviewed (commented):
Looks good. You should also make this change in the `KeyboardEvent` constructor.

```diff
diff --git a/Source/WebCore/dom/KeyboardEvent.cpp b/Source/WebCore/dom/KeyboardEvent.cpp
index 9ab348f6b76a..6ec2fabd75af 100644
--- a/Source/WebCore/dom/KeyboardEvent.cpp
+++ b/Source/WebCore/dom/KeyboardEvent.cpp
@@ -134,7 +134,7 @@ inline KeyboardEvent::KeyboardEvent(const AtomString& eventType, const Init& ini
     , m_key(initializer.key)
     , m_code(initializer.code)
     , m_keyIdentifier(initializer.keyIdentifier)
-    , m_location(initializer.keyLocation ? *initializer.keyLocation : initializer.location)
+    , m_location(initializer.location)
     , m_repeat(initializer.repeat)
     , m_isComposing(initializer.isComposing)
     , m_charCode(initializer.charCode)
```

aprotyas commented on LayoutTests/fast/events/constructors/keyboard-event-constructor.html:
```diff
@@ -80,40 +80,36 @@
-shouldBe("new KeyboardEvent('eventType', { location: KeyboardEvent.DOM_KEY_LOCATION_LEFT }).keyLocation", "KeyboardEvent.DOM_KEY_LOCATION_LEFT");
+// location is passed.
+// numbers within the unsigned long range.
+shouldBe("new KeyboardEvent('eventType', { location : KeyboardEvent.DOM_KEY_LOCATION_STANDARD }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
```
One more thing, the results diff will be smaller if you make this adjustment to all the cases.

```suggestion
shouldBe("new KeyboardEvent('eventType', { location: KeyboardEvent.DOM_KEY_LOCATION_STANDARD }).location", "KeyboardEvent.DOM_KEY_LOCATION_STANDARD");
```

Ahmad-S792 replied:
Done in [32ebe55](https://github.com/WebKit/WebKit/commit/32ebe55a4c813f1fcb5e822f57195660afc70f62).

Ahmad-S792 committed:
Remove non-standard `KeyboardEvent.keyLocation`

https://bugs.webkit.org/show_bug.cgi?id=267819

Reviewed by Ryosuke Niwa.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium
by removing non-standard KeyboardEvent.keyLocation (alias of 'location').

It is not in web-specification [1].

[1] https://w3c.github.io/uievents/#events-keyboardevents

It was never shipped in Gecko / Firefox while removed in Blink
in 2016 and with commit [2]:

[2] https://chromium.googlesource.com/chromium/src.git/+/263d94f9b2dc4a0a3cf5b3124c1cc8d2163a4410

* Source/WebCore/dom/KeyboardEvent.cpp:
(KeyboardEvent::KeyboardEvent):
* Source/WebCore/dom/KeyboardEvent.h:
* Source/WebCore/dom/KeyboardEvent.idl:
* LayoutTests/fast/events/constructors/keyboard-event-constructor.html: Rebaselined
* LayoutTests/fast/events/constructors/keyboard-event-constructor-expected.txt: Rebaselined
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-control-expected.txt: Rebaselined
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-control-option-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-control-shift-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-meta-control-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-meta-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-meta-option-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-meta-shift-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-option-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-option-shift-expected.txt: Ditto
* LayoutTests/fast/events/ios/key-events-comprehensive/key-events-shift-expected.txt: Ditto

Canonical link: https://commits.webkit.org/273457@main
